### PR TITLE
Error codes

### PR DESCRIPTION
**Pull request by mariane-bekker**

### CODE REVIEW

File: lib/PropertyFilter.js [hunk 1]
```diff
@@ -1,101 +1,135 @@
...
-    for (var key in this._data) {
-        if (this._data.hasOwnProperty(key) && key[0] == '$') {
-            filtered[key] = this._data[key];
+    for (var key in data) {
+        if (data.hasOwnProperty(key) && key[0] == '$') {
+            filtered[key] = data[key];
+        }
+    }
+    // Directly assign top-level props or properties beginning with $
+    if (property.indexOf('$') !== -1 || (-1 === property.indexOf('.') && -1 === property.indexOf('@'))) {
```
Comment (djvirgen) [inline lib/PropertyFilter.js]: The statement `property.indexOf('$') !== -1` would evaluate to true if the `$` character appears anywhere in the key name, e.g. `abc$123`. The previous implementation and the comment indicate this should only happen for properties that begin with `$`.

Comment (djvirgen) [inline lib/PropertyFilter.js]: `(-1 === property.indexOf('.') && -1 === property.indexOf('@'))` why do we require that *both* `.` and `@` are in the prop?

Comment (mariane-bekker) [inline lib/PropertyFilter.js]: ahh good catch!!! will update

Comment (mariane-bekker) [inline lib/PropertyFilter.js]: updated

**Comment by djvirgen** [comment]
🍒

**Pull request merged**